File: bmi.c
// Compilation Instructions
// gcc bmi.c && ./a.out

#include <stdio.h>
#include <stdlib.h>

#define SIZE 10

int main() {

  char weight_c[SIZE], height_c[SIZE];
  double weight, height, bmi;

  printf("Enter you weight: ");
  fgets(weight_c, SIZE, stdin);
  weight = atof(weight_c);

  printf("Enter you height: ");
  fgets(height_c, SIZE, stdin);
  height = atof(height_c);

  bmi = weight / (height * height);

  printf("weight is %2f, height is %2f\n", weight, height);
  printf("You bmi is %2f\n", bmi);
}

File: bmi.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	stdin := bufio.NewReader(os.Stdin)

	fmt.Print("Enter your weight (kg): ")
	weightString, err := stdin.ReadString('\n')
	handle(err)
	weight, err := strconv.ParseFloat(strings.TrimSpace(weightString), 64)
	handle(err)

	fmt.Print("Enter your height (m): ")
	heightString, err := stdin.ReadString('\n')
	handle(err)
	height, err := strconv.ParseFloat(strings.TrimSpace(heightString), 64)
	handle(err)

	bmi := weight / (height * height)
	fmt.Printf("Your BMI is %.2f\n!", bmi)
}

func handle(err error) {
	if err != nil {
		fmt.Printf("error: %v\n", err)
		os.Exit(1)
	}
}
